core/dbt/cli/params.py

```diff
@@ -0,0 +1,383 @@
+# This will need to be fixed before use in the main codebase and communicated as a change to the community!
+print = click.option(
+    "--print/--no-print",
+    envvar="DBT_PRINT",
+    help="Output all {{ print() }} macro calls.",
+    default=True,
+)
+
+printer_width = click.option(
+    "--printer-width",
+    envvar="DBT_PRINTER_WIDTH",
+    help="Sets the width of terminal output",
+    type=click.INT,
+    default=80,
+)
+
+profile = click.option(
+    "--profile",
+    envvar=None,
+    help="Which profile to load. Overrides setting in dbt_project.yml.",
+)
+
+profiles_dir = click.option(
+    "--profiles-dir",
+    envvar="DBT_PROFILES_DIR",
+    help=f"Which directory to look in for the profiles.yml file. Default = {PurePath.joinpath(Path.home(), '.dbt')}",
+    default=PurePath.joinpath(Path.home(), ".dbt"),
+    type=click.Path(
+        exists=True,
+    ),
+)
+
+project_dir = click.option(
+    "--project-dir",
+    envvar=None,
+    help="Which directory to look in for the dbt_project.yml file. Default is the current working directory and its parents.",
+    default=Path.cwd(),
+    type=click.Path(exists=True),
+)
+
+quiet = click.option(
+    "--quiet/--no-quiet",
+    envvar="DBT_QUIET",
+    help="Suppress all non-error logging to stdout. Does not affect {{ print() }} macro calls.",
+)
+
+record_timing_info = click.option(
+    "--record-timing-info",
+    "-r",
+    envvar=None,
+    help="When this option is passed, dbt will output low-level timing stats to the specified file. Example: `--record-timing-info output.profile`",
+    is_flag=True,
+)
+
+resource_type = click.option(
+    "--resource-type",
+    envvar=None,
+    help="TODO: No current help text",
+    type=click.Choice(
+        [
+            "metric",
+            "source",
+            "analysis",
+            "model",
+            "test",
+            "exposure",
+            "snapshot",
+            "seed",
+            "default",
+            "all",
+        ],
+        case_sensitive=False,
+    ),
+    default="default",
+)
+
+selector = click.option(
+    "--selector", envvar=None, help="The selector name to use, as defined in selectors.yml"
+)
+
+show = click.option(
+    "--show", envvar=None, help="Show a sample of the loaded data in the terminal", is_flag=True
+)
+
+skip_profile_setup = click.option(
+    "--skip-profile-setup", "-s", envvar=None, help="Skip interative profile setup.", is_flag=True
+)
+
+# TODO:  The env var and name (reflected in flags) are corrections!
+# The original name was `ARTIFACT_STATE_PATH` and used the env var `DBT_ARTIFACT_STATE_PATH`.
+# Both of which break existing naming conventions.
+# This will need to be fixed before use in the main codebase and communicated as a change to the community!
+state = click.option(
+    "--state",
+    envvar="DBT_STATE",
+    help="If set, use the given directory as the source for json files to compare with this project.",
+    type=Truthy(),
```

Comment:
Should `--state` use `type=click.Path()`?

Reply:
Yeah, it probably should.  Technically it's both Truthy and a path, but is doesn't _have_ to be truthy... it's just an ugly pattern we fell into at some point.

Updated!